src/context/wishlist-context/wishlist-context.js

```diff
@@ -0,0 +1,95 @@
+import { useContext, useState, createContext } from "react";
+import { useEffect } from "react";
+import { useToast } from "../toast-context/toast-context";
+import axios from "axios";
+import { useAuth } from "../auth-context/auth-context";
+
+const WishListContext = createContext();
+
+const useWishList = () => useContext(WishListContext);
+
+function WishListProvider({ children }) {
+  const { encodedToken } = useAuth();
+
+  const [wishList, setWishList] = useState([]);
+  const { dispatch } = useToast();
+
+  useEffect(() => {
+    if (encodedToken) {
+      (async function () {
+        try {
+          const wishListResponse = await axios.get("/api/user/wishlist", {
+            headers: {
+              authorization: encodedToken,
+            },
+          });
+
+          if (wishListResponse.status === 200) {
+            setWishList(wishListResponse.data.wishlist);
+          }
+        } catch (error) {
+          console.error(error.response);
+        }
```

Comment:
Suggestion: You could create a common api call function to avoid writing same code again and again.